libqtile/core/manager.py

```diff
@@ -527,25 +527,40 @@ def current_group(self):
     def current_window(self):
         return self.current_screen.group.current_window
 
-    def add_strut(self, strut):
+    def add_strut(self, strut, screen):
         from libqtile.bar import Bar, Gap
 
+        if any(e is None for e in [screen, screen.x, screen.y, screen.height, screen.width]):
```

Comment:
This is just a bit easier to read:
```suggestion
        if not all([screen, screen.x, screen.y, screen.height, screen.width]):
```
or even:
```suggestion
        if None in [screen, screen.x, screen.y, screen.height, screen.width]:
```

Reply:
`if None in [...]` is probably more explicit. If `screen.x == 0`, then `not all([...])` would be `True = not False` whereas `None` might not be in `[...]`. We probably don't want to return if one item is `0`, right?

Reply:
Ah yep good spot, `if None in` is definitely the way to go.

Reply:
Yes, the second one looks better.